modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/KeyValueStorage.java

```diff
@@ -292,4 +294,11 @@ public interface KeyValueStorage extends ManuallyCloseable {
      * @return Timestamp corresponding to the revision.
      */
     HybridTimestamp timestampByRevision(long revision);
+
+    /**
+     * Sets the revision listener. This is needed only for the recovery, after that listener must be set to {@code null}.
+     *
+     * @param listener Revision listener.
+     */
+    void setRevisionListener(@Nullable LongConsumer listener);
```

Comment:
I think that the name is not quite suitable, it is better to indicate that this is a listener on recovery, plus it would be better to use the new interface instead of `LongConsumer`,
What is the meaning of **null** listener?